### PR TITLE
update opentelemetry external dependencies

### PR DESCRIPTION
**Pull request by kryalama**
update opentelemetry-sdk to version 1.11.0
update opentelemetry-api to version 1.11.0
update opentelemetry-exporter-logging to version 1.11.0
update opentelemetry-exporter-jaeger to version 1.11.0

### CODE REVIEW

File: eng/versioning/external_dependencies.txt [hunk 1]
```diff
@@ -183,8 +183,8 @@ com.microsoft.azure:azure-storage;8.0.0
 com.microsoft.azure:msal4j;1.11.0
 com.microsoft.azure:msal4j-persistence-extension;1.1.0
 com.sun.activation:jakarta.activation;1.2.2
-io.opentelemetry:opentelemetry-api;1.0.0
-io.opentelemetry:opentelemetry-sdk;1.0.0
+io.opentelemetry:opentelemetry-api;1.10.0
+io.opentelemetry:opentelemetry-sdk;1.10.0
 io.opentelemetry:opentelemetry-exporter-logging;1.0.0
 io.opentelemetry:opentelemetry-exporter-jaeger;1.0.0
```
Comment (kryalama) [inline eng/versioning/external_dependencies.txt:189]: @trask should I also update these 2 packages?

Comment (trask) [reply eng/versioning/external_dependencies.txt:189]: yes 👍

File: eng/versioning/external_dependencies.txt [hunk 2]
```diff
@@ -183,10 +183,10 @@ com.microsoft.azure:azure-storage;8.0.0
...
 com.microsoft.azure:msal4j-persistence-extension;1.1.0
 com.sun.activation:jakarta.activation;1.2.2
-io.opentelemetry:opentelemetry-api;1.0.0
-io.opentelemetry:opentelemetry-sdk;1.0.0
-io.opentelemetry:opentelemetry-exporter-logging;1.0.0
-io.opentelemetry:opentelemetry-exporter-jaeger;1.0.0
+io.opentelemetry:opentelemetry-api;1.10.0
+io.opentelemetry:opentelemetry-sdk;1.10.0
+io.opentelemetry:opentelemetry-exporter-logging;1.10.0
+io.opentelemetry:opentelemetry-exporter-jaeger;1.10.0
```
Comment (trask) [inline eng/versioning/external_dependencies.txt:189]: let's bump to latest now 1.11.0

File: eng/versioning/external_dependencies.txt [hunk 3]
```diff
@@ -225,6 +225,7 @@ groovy3_org.spockframework:spock-core;2.0-groovy-3.0
 org.testng:testng;7.3.0
 uk.org.lidalia:slf4j-test;1.2.0
 com.google.truth:truth;1.1.3
+io.grpc:grpc-api;1.44.0
```
Comment (trask) [inline eng/versioning/external_dependencies.txt:228]: the otel exporters no longer have a dependency on grpc library, hoping we can undo this change and below?

Comment (kryalama) [reply eng/versioning/external_dependencies.txt:228]: The examples in the repo uses ManagedChannel from grpc library. Had to add this as test dependency

Comment (trask) [reply eng/versioning/external_dependencies.txt:228]: let's change the examples, otel-java is not recommending to bring your own ManagedChannel anymore, as it will use OkHttp for transport now if you do not

**Review by trask — APPROVED** [review]

**Comment by srnagar** [comment]
Looks good to me. The PR did not trigger builds for exporter and attestation. Please run those pipelines too before merging.

**Review by srnagar — APPROVED** [review]
